user-operator/src/main/java/io/strimzi/operator/user/operator/cache/AclCache.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.user.operator.cache;
+
+import io.strimzi.operator.user.model.acl.SimpleAclRule;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.common.KafkaFuture;
+import org.apache.kafka.common.acl.AclBinding;
+import org.apache.kafka.common.acl.AclBindingFilter;
+import org.apache.kafka.common.security.auth.KafkaPrincipal;
+import org.apache.kafka.common.utils.SecurityUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * A periodically updated ACL Cache for keeping the ACLs locally and avoid querying Kafka
+ */
+public class AclCache extends AbstractCache<Set<SimpleAclRule>> {
+    private final static Logger LOGGER = LogManager.getLogger(AclCache.class);
+
+    private final Admin adminClient;
+
+    public AclCache(Admin adminClient, long refreshIntervalMs) {
+        super("ACL", refreshIntervalMs);
+        this.adminClient = adminClient;
+    }
+
+    /**
+     * Loads the ACL rules from Kafka for all users
+     *
+     * @return  ConcurrentHashMap with all users and their ACLs
+     */
+    @Override
+    protected ConcurrentHashMap<String, Set<SimpleAclRule>> loadCache() {
+        KafkaFuture<Collection<AclBinding>> futureAcls = adminClient.describeAcls(AclBindingFilter.ANY).values();
+
+        try {
+            Collection<AclBinding> aclsBindings = futureAcls.get(1, TimeUnit.MINUTES);
+            ConcurrentHashMap<String, Set<SimpleAclRule>> map = new ConcurrentHashMap<>();
```

Comment:
I added this to the Quotas and ScramShaCredentials cache. But not 100% if it makes sense for ACLs.  Kafka gives us the ACLs as one big collection. But we store them in the cache per user. So if you have at average 5 ACL rules per user, the collection might have a 100 ACL rules, but the map might have only 20 keys. So if we size the map, it can actually be fairly oversized. Do you think it would still make sense?

Reply:
Right yeah. Maybe dividing by 5 is a good rule of thumb though (better than the default capacity).